src/main/java/uk/gov/laa/ccms/caab/service/ClientService.java

```diff
@@ -77,16 +78,19 @@ public Mono<ClientStatus> getClientStatus(String transactionId, String loginId,
   /**
    * Creates a client based on a given client details.
    *
-   * @param clientDetails         The client's details.
+   * @param clientFlowFormData    The client's details.
    * @param user                  The user.
    * @return A Mono wrapping the ClientCreated transaction id.
    */
   public Mono<ClientCreated> createClient(
-      uk.gov.laa.ccms.caab.bean.ClientDetails clientDetails,
+      ClientFlowFormData clientFlowFormData,
       UserDetail user) {
 
-    ReflectionUtils.nullifyStrings(clientDetails);
-    ClientDetail clientDetail = clientDetailsMapper.toSoaClientDetail(clientDetails);
+    ReflectionUtils.nullifyStrings(clientFlowFormData.getBasicDetails());
```

Comment:
This need to nullify fields feels like a bug/limitation in SOA. Should these nullify calls not be pushed into the soaApiClient?